### PR TITLE
Dunitz/czi hosted test server

### PR DESCRIPTION
**Pull request by MDunitz**
#### Reviewers
**Functional:** 
- @Bento007 
**Readability:** 
- @maniarathi 
---

## Changes
- modify unit tests to use test flask server instead of cellxgene package to better simulate the production environment

### CODE REVIEW

File: backend/czi_hosted/app/app.py [hunk 1]
```diff
@@ -475,6 +475,6 @@ def __init__(self, app_config):
 
         auth = server_config.auth
         self.app.auth = auth
-        if auth.requires_client_login():
+        if auth and auth.requires_client_login():
```
[the remark this answers is not in the document]
Comment (MDunitz) [reply backend/czi_hosted/app/app.py:478]: sometimes auth is set to None in the config. When that happens if auth.requires_client_login() throws an error

File: backend/czi_hosted/common/config/app_config.py [hunk 2]
```diff
@@ -69,21 +69,21 @@ def check_config(self):
 
     def update_server_config(self, **kw):
         self.server_config.update(**kw)
-        self.is_complete = False
+        self.is_completed = False
```
[the remark this answers is not in the document]
Comment (MDunitz) [reply backend/czi_hosted/common/config/app_config.py:72]: They should all be `is_completed` I noticed that there were instances of both throughout the file so I tried to update them all to is_completed. Let me know where you are seeing `is_complete`

File: backend/test/test_czi_hosted/unit/auth/test_auth.py [hunk 3]
```diff
@@ -18,13 +17,13 @@ def test_auth_none(self):
...
-            session = requests.Session()
-            config = session.get(f"{server}/d/pbmc3k.cxg/api/v0.2/config").json()
-            userinfo = session.get(f"{server}/d/pbmc3k.cxg/api/v0.2/userinfo").json()
-            self.assertNotIn("authentication", config["config"])
-            self.assertIsNone(userinfo)
+        server= self.create_app(app_config)
+        server.testing = True
+        session = server.test_client()
+        config = json.loads(session.get(f"{self.TEST_URL_BASE}config").data)
+        userinfo = json.loads(session.get(f"{self.TEST_URL_BASE}userinfo").data)
```
[the remark this answers is not in the document]
Comment (MDunitz) [reply backend/test/test_czi_hosted/unit/auth/test_auth.py:24]: We need to set the config with particular vars before creating the app, so there is a lot of repeat code between different tests that cant easily be pulled out

File: backend/test/test_czi_hosted/unit/common/config/test_server_config.py [hunk 4]
```diff
@@ -147,30 +146,31 @@ def test_handle_data_source__errors_when_passed_zero_or_two_dataroots(self):
...
     def test_get_api_base_url_works(self):
-
         # test the api_base_url feature, and that it can contain a path
         config = AppConfig()
         backend_port = find_available_port("localhost", 10000)
         config.update_server_config(
             app__flask_secret_key="secret",
             app__api_base_url=f"http://localhost:{backend_port}/additional/path",
             multi_dataset__dataroot=f"{PROJECT_ROOT}/example-dataset",
+            multi_dataset__allowed_matrix_types=["h5ad"],
```
[the remark this answers is not in the document]
Comment (MDunitz) [reply backend/test/test_czi_hosted/unit/common/config/test_server_config.py]: good catch - it should be cxg (all of hosted should be cxg) but it doesnt seem to be the underlying cause of the failure

**Review by Bento007 — APPROVED** [review]

**Pull request merged**